src/MediatR/Mediator.cs

```diff
@@ -73,20 +88,65 @@ public void Publish(INotification notification)
 
         public Task PublishAsync(IAsyncNotification notification)
         {

```

Comment:
To not introduce breaking change, I'd make this an overload having the default behavior.